<?xml version="1.0"?>
<launch>
  ############################# Global Param ###################################

  <arg name="debug" default="false" />

  <!-- estimate mode -->
  <arg name="state_mode" value="0" />
  #GROUND_TRUTH = 0
  #EGOMOTION_ESTIMATE = 1
  <!-- for the outdoor experiment using magnetic, should not be here -->
  <arg name="only_imu_yaw" value="false" />

  <!-- flight control mode: should only be one true flag -->
  <arg name="pos_control_mode" value="true" /> <!-- pos mode -->
  <arg name="vel_control_mode" value="false" />  <!-- vel mode -->
  <arg name="att_control_mode" value="false" />  <!-- att mode -->

  ############## Motor Info ############################
  <rosparam file="$(find hydrus_transform_control)/config/hydrus3/MotorInfo.yaml" command="load" /> <!-- basic motor info -->

  ############### Sensors #############################
  <include file="$(find hydrus_transform_control)/launch/hydrus3/hydrus3_sensors.launch" >
    <arg name="imu_board" value="1" /> <!-- 1: d imu board, 0: kduino board -->
  </include >

  ############################# Drone Platform ###################################
  <node pkg="aerial_robot_base" type="aerial_robot_base_node" name="aerial_robot_base_node"  launch-prefix="$(arg launch_prefix)" output="screen" > 

    ############### Basic Param  #####################
    <param if="$(arg pos_control_mode)" name="navigator/xy_control_mode" type="int" value="0"/>
    <param if="$(arg vel_control_mode)" name="navigator/xy_control_mode" type="int" value="3"/>
    <param if="$(arg att_control_mode)" name="navigator/xy_control_mode" type="int" value="4"/>

    <param name="estimator/state_mode" value= "$(state_mode)" />
    <param name="estimator/only_imu_yaw" value="$(arg only_imu_yaw)"/>

    <param name="rx_loop_rate" type="double" value="110"/>
    <param name="tx_loop_rate" type="double" value="40"/>
    <param name="tf_pub_loop_rate" type="double" value="120"/>

    ############### Sensor Fusion ######################
    <rosparam file="$(find hydrus_transform_control)/config/hydrus3/SensorFusion.yaml" command="load" />
    ############### PID Control ######################
    <rosparam file="$(find hydrus_transform_control)/config/hydrus3/PidControlConfig.yaml" command="load" />

v    ############### Keyboard Navigation ##############
    <rosparam file="$(find hydrus_transform_control)/config/hydrus3/TeleopNavigationConfig.yaml" command="load" />
  </node>

  ################################# Dynamixel  ######################################
  <include file="$(find hydrus_transform_control)/launch/hydrus3/hydrus3_dynamixel_bridge.launch" />  

  ################################# Robot Model  ######################################
  <include file="$(find hydrus_transform_control)/launch/hydrus3/hydrus3_robot_model.launch" />  

  ################################# Transform Control  ######################################
  <node pkg="hydrus_transform_control" type="transform_control_node" name="transform_control_node"  output="screen" > 
    <rosparam file="$(find hydrus_transform_control)/config/hydrus3/TransformControl.yaml" command="load" />

  </node>

</launch>



